projects/observability/src/shared/components/iframe/iframe.component.ts

```diff
@@ -0,0 +1,20 @@
+import { ChangeDetectionStrategy, Component, Input, OnInit } from '@angular/core';
+import { DomSanitizer, SafeResourceUrl } from '@angular/platform-browser';
+
+@Component({
+  selector: 'ht-iframe',
+  template: `<iframe width="100%" height="100%" frameBorder="0" [src]="urlSafe"></iframe>`,
+  changeDetection: ChangeDetectionStrategy.OnPush,
+  styleUrls: ['./iframe.component.scss']
+})
+export class IFrameComponent implements OnInit {
+  @Input() public source: string = '';
+  @Input() public title?: string;
+
+  public urlSafe?: SafeResourceUrl;
+  public constructor(public sanitizer: DomSanitizer) {}
+
+  public ngOnInit(): void {
+    this.urlSafe = this.sanitizer.bypassSecurityTrustResourceUrl(this.source);
```

Comment:
we probably would not want to bypass security like this inside a dumb component - would leave it to the place where the URL originates to decide whether it's safe to do this, and instead take in a SafeResourceUrl rather than a string.

Reply:
moved to  widget